src/pages/iou/WaypointEditor.js

```diff
@@ -0,0 +1,146 @@
+        if (props.network.isOffline && waypointValue && ValidationUtils.isValidAddress(waypointValue)) {
+            errors[`waypoint${waypointIndex}`] = 'bankAccount.error.address';
+        }
+
+        return errors;
+    };
+
+    const onSubmit = (values) => {
+        // Allows letting you set a waypoint to an empty value
+        if (!values[`waypoint${waypointIndex}`]) {
+            DistanceRequest.saveWaypoint(transactionID, waypointIndex, null);
+        }
+
+        // While the user is offline, the auto-complete address search will not work
+        // Therefore, we're going to save the waypoint as just the address, and the lat/long will be filled in on the backend
+        if (props.network.isOffline && values[`waypoint${waypointIndex}`]) {
+            const waypoint = {
+                address: values[`waypoint${waypointIndex}`],
+            };
+
+            DistanceRequest.saveWaypoint(transactionID, waypointIndex, waypoint);
+        }
+
+        Navigation.navigate(ROUTES.getMoneyRequestDistanceTabRoute(iouType));
+    };
+
+    const onPress = (values) => {
+        const waypoint = {
+            lat: values.lat,
+            lng: values.lng,
+            address: values.address,
+        };
+
+        DistanceRequest.saveWaypoint(transactionID, waypointIndex, waypoint);
+        Navigation.navigate(ROUTES.getMoneyRequestDistanceTabRoute(iouType));
+    };
+
+    return (
+        <ScreenWrapper includeSafeAreaPaddingBottom={false}>
+            <HeaderWithBackButton
+                title="Waypoint Editor"
+                shouldShowBackButton
+                onBackButtonPress={() => {
+                    Navigation.navigate(ROUTES.getMoneyRequestDistanceTabRoute(iouType));
+                }}
+            />
+            <Form
+                style={[styles.flexGrow1, styles.mh5]}
+                formID={ONYXKEYS.FORMS.WAYPOINT_FORM}
+                enabledWhenOffline
+                validate={validate}
+                onSubmit={onSubmit}
+                submitButtonText={props.translate('common.save')}
+            >
+                <View>
+                    <AddressSearch
+                        inputID={`waypoint${waypointIndex}`}
+                        containerStyles={[styles.mt4]}
+                        label={props.translate('distance.address')}
+                        shouldSaveDraft
+                        onPress={onPress}
+                        maxInputLength={CONST.FORM_CHARACTER_LIMIT}
+                        renamedInputKeys={{
+                            address: `waypoint${waypointIndex}`,
+                            city: null,
+                            country: null,
+                            street: null,
+                            street2: null,
+                            zipCode: null,
+                            lat: null,
+                            lng: null,
+                            state: null,
```

Comment:
These `props` are not defined here
https://github.com/Expensify/App/blob/a2096c8ecf6858d4efe4c4fa47f72f54872ee798/src/components/AddressSearch/index.js#L59-L64

Can we just define there set default values there once?